### ISSUE TITLE
az aro delete fails with a traceback when provided with invalid resource group and cluster names

### ISSUE THREAD

**Issue by mgahagan73** [Azure/azure-cli#18501]
## Describe the bug

**Command Name**
`az aro delete`

**Errors:**
```
The command failed with an unexpected error. Here is the traceback:
local variable 'oc' referenced before assignment
Traceback (most recent call last):
  File "/usr/lib64/az/lib/python3.6/site-packages/knack/cli.py", line 231, in invoke
    cmd_result = self.invocation.execute(args)
  File "/usr/lib64/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 657, in execute
    raise ex
  File "/usr/lib64/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 720, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
  File "/usr/lib64/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 691, in _run_job
    result = cmd_copy(params)
  File "/usr/lib64/az/lib/python3.6/site-packages/azure/cli/core/commands/__init__.py", line 328, in __call__
    return self.handler(*args, **kwargs)
  File "/usr/lib64/az/lib/python3.6/site-packages/azure/cli/core/commands/command_operation.py", line 121, in handler
    return op(**command_args)
  File "/usr/lib64/az/lib/python3.6/site-packages/azure/cli/command_modules/aro/custom.py", line 164, in aro_delete
    ensure_resource_permissions(cmd.cli_ctx, oc, False, [rp_client_sp.object_id])
UnboundLocalError: local variable 'oc' referenced before assignment

```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az aro delete --resource-group {} -n {}`

## Environment Summary
```
Linux-5.12.9-300.fc34.x86_64-x86_64-with-glibc2.33
Python 3.9.5
Installer: RPM

azure-cli 2.25.0

I have reproduced the same traceback with both python 3.6 and python 3.9 on a Fedora 34 system

```

**Comment by mgahagan73** [comment]
It turns out this failure only occurs when the cli is given a non-existant resource group name and cluster name

**Comment by yonzhan** [comment]
route to service team

**Comment by bennerv** [comment]
Resolved with #18546 merged in.  Can close @yonzhan
